src/traces/candlestick/attributes.js

```diff
@@ -0,0 +1,49 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+var Lib = require('../../lib');
+var OHLCattrs = require('../ohlc/attributes');
+var boxAttrs = require('../box/attributes');
+
+var directionAttrs = {
+    visible: {
```

Comment:
That looks like a good default, but we had better let users override the name separately for the two halves as well, including with a blank string in case they want the legend to have both symbols but only one label.

We could ask the plotly.py candlestick authors ( @chriddyp @cldougl ? tools.py is getting a little out of hand!) if there was a specific request for separately controlling the two visibilities.